tests/functional/test_states.py

```diff
@@ -0,0 +1,93 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# todo add new DSL variant for job and multijob
+# todo add multistate tests for job and multijob
+
+# pylint:disable=redefined-outer-name, no-self-use
+import allure
+from adcm_client.objects import ADCMClient
+from adcm_pytest_plugin.utils import fixture_parametrized_by_data_subdirs
+from tests.functional.plugin_utils import build_objects_checker
+
+
+ACTION_NAME = 'state_changing_action'
+
+
+@fixture_parametrized_by_data_subdirs(__file__, 'cluster_and_service')
+def cluster(sdk_client_fs: ADCMClient, request):
+    bundle = sdk_client_fs.upload_from_fs(request.param)
+    cluster = bundle.cluster_create(name=bundle.name)
+    cluster.service_add(name='first_srv')
+    check_objects_state_changed = build_objects_checker(
+        field_name='State',
+        changed=cluster.name,
+        extractor=lambda obj: obj.state,
+    )
+    return cluster, check_objects_state_changed
+
+
+class TestClusterRelatedObjects:
+    def test_cluster_state_after_action(self, sdk_client_fs: ADCMClient, cluster):
+        cluster_obj, check_objects_state_changed = cluster
+        object_to_be_changed = cluster_obj
+        with check_objects_state_changed(sdk_client_fs, {object_to_be_changed}), allure.step(
+            f'Run cluster action: {ACTION_NAME}'
+        ):
+            object_to_be_changed.action(name=ACTION_NAME).run().wait()
+
+    def test_service_state_after_action(self, sdk_client_fs: ADCMClient, cluster):
+        cluster_obj, check_objects_state_changed = cluster
+        object_to_be_changed = cluster_obj.service(name='first_srv')
+        with check_objects_state_changed(sdk_client_fs, {object_to_be_changed}), allure.step(
+            f'Run service action: {ACTION_NAME}'
+        ):
+            object_to_be_changed.action(name=ACTION_NAME).run().wait()
+
+    def test_component_state_after_action(self, sdk_client_fs: ADCMClient, cluster):
+        cluster_obj, check_objects_state_changed = cluster
+        object_to_be_changed = cluster_obj.service(name='first_srv').component(name='first_cmp')
+        with check_objects_state_changed(sdk_client_fs, {object_to_be_changed}), allure.step(
+            f'Run component action: {ACTION_NAME}'
+        ):
+            object_to_be_changed.action(name=ACTION_NAME).run().wait()
+
+
+@fixture_parametrized_by_data_subdirs(__file__, 'provider_and_host')
+def provider(sdk_client_fs: ADCMClient, request):
+    bundle = sdk_client_fs.upload_from_fs(request.param)
+    provider = bundle.provider_create(name=bundle.name)
+    provider.host_create(fqdn=bundle.name)
+    check_objects_state_changed = build_objects_checker(
+        field_name='State',
+        changed=bundle.name,
+        extractor=lambda obj: obj.state,
+    )
```

Comment:
In tests on plugins (adcm_config, adcm_state, adcm_multi_state_*set) multiple entities are created (e.g. two providers with two hosts on each) to ensure that they aren't affected by changes. And it's where objects checker is good at. If it's your case, consider creating multiple ADCM objects of one type in `provider` and `cluster` fixtures

Reply:
Fixed